src/components/EntityOpsLevelMaturityContent.tsx

```diff
@@ -59,10 +60,92 @@ export const EntityOpsLevelMaturityContent = () => {
+  const levels = opsLevelData.account.rubric?.levels?.nodes;
+  const levelCategories = opsLevelData.account.service.maturityReport?.categoryBreakdown;
+  const checkResultsByLevel = opsLevelData.account.service.serviceStats?.rubric?.checkResults?.byLevel?.nodes;
+  const scorecards = opsLevelData.account.service.serviceStats?.scorecards?.nodes;
+  // const checkStats = opsLevelData.account.service.checkStats;
+  const scorecardCategories = scorecards ? scorecards.reduce(
+    (scoreCardAccumulator: LevelCategory[], currentScorecard) => {
+      if (!currentScorecard.categories?.edges) return scoreCardAccumulator;
+      return [
+        ...scoreCardAccumulator,
+        ...currentScorecard.categories.edges.reduce(
+          (nodeAccumulator: LevelCategory[], currentEdge): LevelCategory[] => [
+            ...nodeAccumulator,
+            {
+              category: {name: currentEdge.node?.name ?? ''},
+              level: currentEdge.level?.name ? {name: currentEdge.level?.name} : null,
+              rollsUp:
+                  currentScorecard?.scorecard?.affectsOverallServiceLevels,
+            },
+          ],
+          [],
+        ),
+      ];
+    },
+    [],
+  ).sort((a, b) => (a.category.name < b.category.name ? -1 : 1)) : undefined;
+  // const scorecardCategories: LevelCategory[] = []
+
+
+  function checksByLevelIncludingScorecards() {
+    const result = cloneDeep(checkResultsByLevel);
+
+    result?.forEach((checkResults) => {
+      // Use level's 'index' field b/c we don't have level ID here. Note: 'index' *is not* the same as array index
+      const levelIndex = checkResults.level.index;
+
+      scorecards?.forEach((scorecard) => {
+        const entry = scorecard.checkResults?.byLevel?.nodes?.find(
+          (node) => node.level.index === levelIndex,
+        );
+        if (!entry) return;
+
+        entry.items.nodes.forEach((node) => {
+          node.check.isScorecardCheck = true;
+        });
+        checkResults.items.nodes = [
+          ...checkResults.items.nodes,
+          ...entry.items.nodes,
+        ];
+      })
+    })
+    return result || [];
+  }
+
+  const allCheckResultsByLevel = checksByLevelIncludingScorecards()
+
+  function totalChecks() {
+    return allCheckResultsByLevel?.reduce(
+      (accumulator, checkByLevel) =>
+        accumulator + checkByLevel.items.nodes.length,
+      0,
+    );
+  }
+
+  const ResultStatusEnum = Object.freeze({
+    FAILED: "failed",
+    PENDING: "pending",
+    PASSED: "passed",
+    UPCOMING_FAILED: "upcoming_failed",
+    UPCOMING_PENDING: "upcoming_pending",
+    UPCOMING_PASSED: "upcoming_passed",
+  });
+
+  function totalPassingChecks() {
+    return allCheckResultsByLevel.reduce(
+      (accumulator, checkByLevel) =>
+        accumulator +
+        checkByLevel.items.nodes.reduce(
+          (passingChecks, check) =>
+            passingChecks +
+            (check.status === ResultStatusEnum.PASSED  ? 1 : 0),
```

Comment:
Do we need to do anything here with checks that have been configured to be passing if pending? Or is that handled by the OL backend?

Reply:
Genuinely unsure, but this matches the behavior in the OpsLevel app today, so I think it's still solid to have them matching now.

I believe those get passed with `"upcoming_passed"` and don't count towards the total, but I am very unsure on what the ideal logic should be.

Reply:
If it's consistent with what the OL app is doing that's good enough for me 👍